sale_manual_delivery/wizard/manual_proc.py

```diff
@@ -123,8 +128,9 @@ def record_picking(self):
                     vals["date_planned"] = date_planned
                     vals["carrier_id"] = carrier_id.id
                     vals["partner_dest_id"] = wizard.partner_id.id
-
-                    proc_group_obj.run(
+                    if wizard.route_id:
+                        vals["route_ids"] = wizard.route_id
+                    proc_group_obj.with_context(vals=vals).run(
```

Comment:
To handle the carrier, instead of overloading `_get_new_picking_values` to get the carrier in `vals`, why not update the current order.carrier_id with the carrier set in the wizard? Then the module `delivery` will get the carrier as usual from the order (https://github.com/odoo/odoo/blob/11.0/addons/delivery/models/stock_move.py#L26). What do you think?

Reply:
:thinking:  Good point! and same for the route

Reply:
Not sure about the route, setting it in `vals` seems OK, it is then passed in `run(...)` and handled later when generating the move. The thing we don't need is to pass `vals` in the context.

Reply:
For some reason it was not working when setting it in vals only. In any case, updating the order should work.